From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Armbian Build <build@armbian.com>
Date: Sun, 10 May 2026 00:00:00 +0000
Subject: arm64: dts: orangepi-zero3: fix pinctrl error -19 on boot

Similar to H3/H5, the H616/H618 U-Boot shows a benign pinctrl error
because both TF-A/BL31 and U-Boot try to set pinctrl for UART0.
Remove the pinctrl properties from UART0 to prevent the error:

  ns16550_serial serial@5000000: pinctrl_select_state_full:
  uclass_get_device_by_phandle_id: err=-19

The pins are already correctly configured by the earlier boot stages.
---
 dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero3.dtsi | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero3.dtsi b/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero3.dtsi
index 000000000000..000000000000 100644
--- a/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero3.dtsi
+++ b/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero3.dtsi
@@ ... @@
 
 &uart0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_ph_pins>;
 	status = "okay";
 };
-- 
2.35.3
